homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,146 @@
+  The same goes for the friendly names:
+  - sensor.switcher_kis_boiler_schedule0: Boiler Schedule 0
+  ...
+  - sensor.switcher_kis_boiler_schedule7: Boiler Schedule 7
+
+  The sensors state will allways provide the minumum information:
+  - Not configured
+  - Not enabled
+  - Due today at 17:30
+  - Due tommorow at 20:00
+
+- Register the following services:
+  - switcher_kis.create_schedule - takes the following mandatory arguments:
+    {'start_time': '19:30', 'end_time': '20:30', 'recurring': false}
+    if recurring is set to true, you must provide an extra argument:
+    {'days': ['Monday', 'Wednesday', 'Saturday']}
+  - switcher_kis.delete_schedule - takes 1 int argument {'schedule_id': 3'}
+  - switcher_kis.enable_schedule - takes 1 int argument {'schedule_id': 0'}
+  - switcher_kis.disable_schedule - takes 1 int argument {'schedule_id': 7'}
+
+"""
+
+from logging import getLogger
+from typing import Dict
+from traceback import format_exc
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_ICON, CONF_NAME
+from homeassistant.helpers import config_validation as cv
+
+
+REQUIREMENTS = ['aioswitcher==2019.2.6']
+
+_LOGGER = getLogger(__name__)
+
+DOMAIN = 'switcher_kis'
+ENTITY_ID_FORMAT = DOMAIN + '_{}'
+
+CONF_PHONE_ID = 'phone_id'
+CONF_DEVICE_ID = 'device_id'
+CONF_DEVICE_PASSWORD = 'device_password'
+CONF_INCLUDE_SCHEDULE_SENSORS = 'include_schedule_sensors'
+CONF_SCHEDULE_SCAN_INTERVAL = 'schedules_scan_interval'
+
+DEFAULT_NAME = 'boiler'
+DEFAULT_SCHEDULES_SCAN_INTERVAL = timedelta(minutes=5)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PHONE_ID): cv.string,
+        vol.Required(CONF_DEVICE_ID): cv.string,
+        vol.Required(CONF_DEVICE_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME,
+                     default=DEFAULT_NAME): cv.slugify,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+        vol.Optional(CONF_ICON): cv.icon,
+        vol.Optional(CONF_INCLUDE_SCHEDULE_SENSORS,
+                     default=False): cv.boolean,
+        vol.Optional(CONF_SCHEDULE_SCAN_INTERVAL,
+                     default=DEFAULT_SCHEDULES_SCAN_INTERVAL): vol.All(
+                         cv.time_period, cv.positive_timedelta)
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass: HomeAssistant, config: Dict) -> bool:
+    """Set up the switcher component."""
+    from ._service_registration import async_reg_services_platforms
+
+    phone_id = config[DOMAIN].get(CONF_PHONE_ID)
+    device_id = config[DOMAIN].get(CONF_DEVICE_ID)
+    device_password = config[DOMAIN].get(CONF_DEVICE_PASSWORD)
+    include_sensors = config[DOMAIN].get(CONF_INCLUDE_SCHEDULE_SENSORS)
+    schedules_scan_interval = config[DOMAIN].get(CONF_SCHEDULE_SCAN_INTERVAL)
+
+    if None in [phone_id, device_id, device_password]:
```

Comment:
If I remember correctly, the `if` block was added because `mypy` "wouldn't shut up" about these values being `Optional[str]` and not `str` which if course was the result of me using the `get` method that returns `None` as default value.
Based on your previous change request, I think think this `if` block might not be needed anyway.